### ISSUE TITLE
[QUESTION] Is the project still maintained?

### ISSUE THREAD

**Issue by jaklan** [tiangolo/typer#300]
Hi all, 

quick question as in the title - is the project actually alive? The last change was commited on 16 Aug 2020, there's no active dev branch, there are 28 open pull requests with no answers, some crucial issues like no support for Click 8.x are not addressed... It looks very alarming. CLI is the crucial part of many apps, so if the project is abandon - it would be really valuable to say it out loud and recommend people to switch back to Click.

Cc: @tiangolo

**Comment by austinbutler** [comment]
https://github.com/tiangolo/typer/issues/263

**Comment by stefanoborini** [comment]
I wonder more what's the point of yet another library to handle CLI interaction when there's already a ton out there that could simply be improved.

We are tired of having to relearn a new API for the same problem again and again.

And no. I don't choose the library. Whatever project I have to modify choses it.

**Comment by ssbarnea** [comment]
While I really liked typer, but as the same time I am glad I used it only one experimental project so far, where it would not be more than a couple of hours to remove it.

Projects relying on a single person do have a high risk of suddenly becoming unmaintained. I know a couple of teams that questioned even maintainability and API stability of click itself. Sometimes is better to stick with argparse.

I hope the situation will change for the better.

**Comment by eteq** [comment]
> I wonder more what's the point of yet another library to handle CLI interaction when there's already a ton out there that could simply be improved.

I for one thing `typer` has a fresh take on this. Yes there are other libraries (and for that matter, the stdlib itself has two!), but `typer`'s reliance on type annotations to make the syntax clean and expressive I think is a different take on this that really isn't "yet another library" - it's a unique solution to an old problem.

With that in mind, I do think the question in this post is important to get answered: the long-term sustainability of this project is critical for people wanting to use it downstream. But I think it's totally legit for you to answer "I need some other contributors/maintainers to help out with maintaining it", @tiangolo. I know that can be hard - I've struggled with it myself as a maintainer of open source packages.  But I hope the interest in this thread and all the PRs show that there's enough excitement with the great work here to make it happen!  And regardless of the answer, thanks for your work on this @tiangolo - new carefully thought-out ideas like this are the beating heart of the Python community, IMHO!

**Comment by septatrix** [comment]
A [recent tweet from @tiangolo](https://twitter.com/tiangolo/status/1403323831155105797) leads me to believe that development might speed up again.

**Comment by tiangolo** [comment]
Hey all! Thanks for the interest and discussion.

Yep, Typer is still maintained, in fact, I just released Typer `0.4.0`, which is compatible with **both** Click 7 and Click 8. :tada:

But of course, it all depends on how much time I have available to dedicate to it.

A lot of my time is spent handling issues, replying to questions, etc. (here and in the other projects). 

If you would like to see faster progress, you could help by answering questions from others, sponsoring, etc. https://typer.tiangolo.com/help-typer/#help-others-with-issues-in-github

**Comment by jaklan** [comment]
@tiangolo great to hear you are still with us! 😄 You haven't mentioned adding new maintainers as an option - what's the reason?

**Comment by bristea** [comment]
The lack of answer to the most important question combined stats from https://github.com/tiangolo/typer/graphs/contributors do show that there is no desire to transform this into a community project. Typer is a wonderful showcase/portfolio project of @tiangolo and is likely to remain like this forever.

The fact that is hosted under a personal account should be another clue. In the end, it would be up to each one to decide if they want to make it a dependency of not. Sadly very few authors mention clearly the scope of their projects as they like the fame but they don't fancy the cost of becoming popular.

**Comment by alexreg** [comment]
@bristea Yes, I believe you're right. It could nevertheless be forked and continued as a project. I have been considering that myself. (It helps that it's licensed under MIT.)

**Comment by RomainBrault** [comment]
@tiangolo it's been a year without update and some important PR are ready to be merge.

Would it be possible to add some maintainer, as suggested, to keep this nice project alive?

I'm sadly considering moving back to click on my projects due to slow maintenance.

**Comment by alexreg** [comment]
@RomainBrault FYI it's been a while since my last release, but if tiangolo is not keen (or able to) maintain this project, I'm happy to accept PRs or even freature requests on [my fork](https://github.com/alexreg/typer-cloup). I've already enhanced functionality a fair bit, and fixed several important bugs.
